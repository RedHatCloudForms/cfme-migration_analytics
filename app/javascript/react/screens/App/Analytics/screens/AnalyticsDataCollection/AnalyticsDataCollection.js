import React from 'react';
import PropTypes from 'prop-types';
import { Spinner, Button } from 'patternfly-react';

const AnalyticsDataCollection = ({ onCancelClick }) => (
  <React.Fragment>
    <div className="large-spinner">
      <Spinner loading size="lg" inline />
      <h3>{__('Collecting inventory data')}</h3>
    </div>
    <Button onClick={onCancelClick}>{__('Cancel')}</Button>
  </React.Fragment>
);

AnalyticsDataCollection.propTypes = {
  onCancelClick: PropTypes.func.isRequired
};

export default AnalyticsDataCollection;
