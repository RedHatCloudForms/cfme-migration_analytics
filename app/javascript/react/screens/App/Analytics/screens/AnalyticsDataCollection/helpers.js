const payloadPathRegex = /\/migration_analytics\/.+$/;

export const getPayloadUrl = payloadPath => {
  const match = payloadPath && payloadPath.match(payloadPathRegex);
  if (!match) return null;
  const { protocol, hostname, port } = window.location;
  const origin = `${protocol}//${hostname}${port ? `:${port}` : ''}`;
  return `${origin}${match[0]}`;
};
