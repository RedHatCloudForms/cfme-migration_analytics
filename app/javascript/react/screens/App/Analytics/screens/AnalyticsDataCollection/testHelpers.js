import { expectOnlySpecificActions } from '../../../../../../common/testReduxHelpers';

export const expectNoActionsOtherThan = (expectedActions, props) => {
  const allActions = ['startInventoryBundleAction', 'fetchBundleTaskAction', 'resetDataCollectionStateAction'];
  expectOnlySpecificActions({ allActions, expectedActions, props });
};
