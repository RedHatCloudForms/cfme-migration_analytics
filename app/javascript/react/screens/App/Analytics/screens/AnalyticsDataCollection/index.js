import { connect } from 'react-redux';
import AnalyticsDataCollection from './AnalyticsDataCollection';
import {
  startInventoryBundleAction,
  fetchBundleTaskAction,
  resetDataCollectionStateAction
} from '../../redux/analyticsActions';
import {
  selectIsPayloadReady,
  selectBundleError,
  selectBundleTaskHref,
  selectIsBundleTaskFinished,
  selectPayloadPath,
  selectPayloadHost,
  selectNumVms
} from '../../redux/analyticsSelectors';
import { getPayloadUrl } from './helpers';

const mapStateToProps = ({ migrationAnalytics: { analytics } }) => {
  const bundleError = selectBundleError(analytics);
  const isBundleTaskFinished = selectIsBundleTaskFinished(analytics);
  const payloadPath = selectPayloadPath(analytics);
  const payloadHost = selectPayloadHost(analytics);
  return {
    selectedProviders: analytics.selectedProviders,
    bundleError,
    bundleTaskHref: selectBundleTaskHref(analytics),
    isFetchingBundleTask: analytics.isFetchingBundleTask,
    isBundleTaskFinished,
    isPayloadReady: selectIsPayloadReady(analytics, { bundleError, isBundleTaskFinished }),
    numVms: selectNumVms(analytics),
    payloadHost,
    payloadPath,
    payloadUrl: getPayloadUrl(payloadPath)
  };
};

export default connect(
  mapStateToProps,
  { startInventoryBundleAction, fetchBundleTaskAction, resetDataCollectionStateAction }
)(AnalyticsDataCollection);
