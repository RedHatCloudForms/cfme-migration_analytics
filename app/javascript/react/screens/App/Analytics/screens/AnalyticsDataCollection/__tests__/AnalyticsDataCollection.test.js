import React from 'react';
import { shallow } from 'enzyme';
import AnalyticsDataCollection from '../AnalyticsDataCollection';

jest.useFakeTimers();

describe('Data collection screen', () => {
  const getBaseProps = () => ({
    onCancelClick: jest.fn(),
    onReturnClick: jest.fn()
  });

  test('renders with a spinner initially', () => {
    expect(shallow(<AnalyticsDataCollection {...getBaseProps()} />)).toMatchSnapshot();
  });

  /*
  test('renders a success message with a download button after payload is loaded', () => {
    const props = {
      ...getBaseProps(),
      isPayloadReady: true,
      numVms: 7,
      payloadPath: 'some/path/to/payload.tgz',
      payloadUrl: 'fake/url/here/payload.tgz'
    };
    const component = shallow(<AnalyticsDataCollection {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('renders a success message with disabled button if there is no download url', () => {
    const props = {
      ...getBaseProps(),
      isPayloadReady: true,
      numVms: 7,
      payloadPath: 'some/path/to/payload.tgz'
    };
    const component = shallow(<AnalyticsDataCollection {...props} />);
    expect(component).toMatchSnapshot();
  });
  */
});
