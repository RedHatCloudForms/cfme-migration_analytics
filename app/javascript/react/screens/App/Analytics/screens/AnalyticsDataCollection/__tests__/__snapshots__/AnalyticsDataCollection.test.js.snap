// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data collection screen renders a success message and download button after payload is loaded 1`] = `
<div
  className="data-collection-status icon-with-content"
>
  <Icon
    className="checkmark"
    name="check"
    type="fa"
  />
  <div>
    <h3>
      Inventory collection complete
    </h3>
    <p>
      7
      Â 
      VMs examined
    </p>
    <div
      className="buttons"
    >
      <div>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={true}
          onClick={[Function]}
        >
          Download Inventory File
        </Button>
      </div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[MockFunction]}
      >
        Return to Summary
      </Button>
    </div>
  </div>
</div>
`;

exports[`Data collection screen renders an error message if data collection failed 1`] = `
<EmptyStateWithButton
  buttonText="Cancel"
  href={null}
  iconName="error-circle-o"
  iconType="pf"
  message="some error here"
  onClick={[MockFunction]}
  title="Failed to collect inventory data"
/>
`;

exports[`Data collection screen renders with a spinner initially and starts the bundle on mount 1`] = `
<div
  className="data-collection-status icon-with-content"
>
  <Spinner
    className=""
    inline={true}
    inverse={false}
    loading={true}
    size="lg"
  />
  <div>
    <h3
      className="beside-spinner"
    >
      Collecting inventory data
    </h3>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      Cancel
    </Button>
  </div>
</div>
`;
